src/gt4py/next/iterator/embedded.py

```diff
@@ -1088,7 +1090,7 @@ def remap(self, index_field: common.ConnectivityField | fbuiltins.FieldOffset) -
         # TODO can be implemented by constructing and ndarray (but do we know of which kind?)
         raise NotImplementedError()
 
-    def restrict(self, item: common.AnyIndexSpec) -> common.Field:
+    def restrict(self, item: common.AnyIndexSpec) -> xtyping.Self:
```

Comment:
```suggestion
    def restrict(self, item: common.AnyIndexSpec) -> Self:
```
and then add the proper import header (ideally, replace the whole `from typing import ...` block by a `from gt4py.eve.xtyping import ...`

Reply:
done